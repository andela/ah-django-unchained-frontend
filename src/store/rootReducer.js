import { combineReducers } from 'redux';
import { signUpReducer } from './modules/signup/index';
import { loginReducer } from './modules/login/index';
import socialAuth from './modules/socialAuth/index';
import { updatepasswordReducer } from './modules/UpdatePassword';
import articleReducer from './modules/getArticles/index';
import { singleArticle } from './modules/articleDetail/index';
import { getDraftReducer } from './modules/drafts/index';
import {
  postArticleReducer,
  imageUploadReducer
} from './modules/createArticle/index';
import { ratingReducer } from './modules/articleRating/index';
import { resetReducer } from './modules/PasswordReset';
import { deleteArticleReducer } from './modules/deleteArticle';
import  userProfile  from './modules/profile/index';
import { likeDislikeArticle } from './modules/likeDislikeArticle/index';

const rootReducer = combineReducers({
  signUpReducer,
  loginReducer,
  socialAuth,
  resetReducer,
  updatepasswordReducer,
  articleReducer,
  singleArticle,
  getDraftReducer,
  postArticleReducer,
  imageUploadReducer,
  ratingReducer,
  deleteArticleReducer,
  userProfile,
  likeDislikeArticle
});

export default rootReducer;
