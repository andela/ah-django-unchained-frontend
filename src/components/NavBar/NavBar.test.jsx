import React from 'react';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import {mapDispatchToProps, NavBar} from './index';
import store from '../../store';

describe('The NavBar', () => {
  let componentWrapper = {};
  let props;
  let dispatch;
  let wrapperInstance;
  beforeEach(() => {
    componentWrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <NavBar />
        </MemoryRouter>
      </Provider>
    );
    props = {
      logoutUser: jest.fn(() => {
        Promise.resolve();
      })
    };
    wrapperInstance = shallow(<NavBar {...props} />).instance();
  });

  it('should match snapshot', () => {
    expect(componentWrapper).toMatchSnapshot();
  });

  it('Should dispatch when logoutuser is called', () => {
    dispatch = jest.fn(() => Promise.resolve());
    props = mapDispatchToProps(dispatch);
    props.logoutUser();
    expect(dispatch).toHaveBeenCalled();
  });

  it('should call dispatch when clearToken is called', () => {
    wrapperInstance.clearToken();
    expect(props.logoutUser).toHaveBeenCalled();
  });
});
