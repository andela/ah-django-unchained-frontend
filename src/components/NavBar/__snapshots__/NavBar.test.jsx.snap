// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The NavBar should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "024jwa",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "024jwa",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <NavBar>
        <div>
          <Navbar
            className="navbar navbar-expand-lg navbar bg-dark navbar-dark "
            expand={false}
            tag="nav"
          >
            <nav
              className="navbar navbar-expand-lg navbar bg-dark navbar-dark  navbar"
            >
              <NavbarBrand
                className="nav-link"
                href="/"
                tag="a"
              >
                <a
                  className="nav-link navbar-brand"
                  href="/"
                >
                  <img
                    alt="Logo"
                    src="logo.png"
                  />
                  Authors Haven
                </a>
              </NavbarBrand>
              <NavbarToggler
                onClick={[Function]}
                tag="button"
                type="button"
              >
                <button
                  className="navbar-toggler"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="navbar-toggler-icon"
                  />
                </button>
              </NavbarToggler>
              <Collapse
                appear={false}
                enter={true}
                exit={true}
                in={false}
                isOpen={false}
                mountOnEnter={false}
                navbar={true}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                tag="div"
                timeout={350}
                unmountOnExit={false}
              >
                <Transition
                  appear={false}
                  enter={true}
                  exit={true}
                  in={false}
                  mountOnEnter={false}
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onEntering={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  onExiting={[Function]}
                  timeout={350}
                  unmountOnExit={false}
                >
                  <div
                    className="collapse navbar-collapse"
                    style={Object {}}
                  >
                    <Nav
                      className="ml-auto"
                      tag="ul"
                      vertical={false}
                    >
                      <ul
                        className="ml-auto nav"
                      >
                        <ul
                          className="container"
                        >
                          <li>
                            <Link
                              className="text-info nav-link"
                              replace={false}
                              to="signup"
                            >
                              <a
                                className="text-info nav-link"
                                href="/signup"
                                onClick={[Function]}
                              >
                                SignUp
                              </a>
                            </Link>
                          </li>
                          <li>
                            <Link
                              className="text-info nav-link"
                              replace={false}
                              to="/login"
                            >
                              <a
                                className="text-info nav-link"
                                href="/login"
                                onClick={[Function]}
                              >
                                Login
                              </a>
                            </Link>
                          </li>
                          <li>
                            <Link
                              className="text-info nav-link"
                              replace={false}
                              to="/draft"
                            >
                              <a
                                className="text-info nav-link"
                                href="/draft"
                                onClick={[Function]}
                              >
                                View Drafts
                              </a>
                            </Link>
                          </li>
                        </ul>
                      </ul>
                    </Nav>
                  </div>
                </Transition>
              </Collapse>
            </nav>
          </Navbar>
        </div>
      </NavBar>
    </Router>
  </MemoryRouter>
</Provider>
`;
